internal/command/group.go

```diff
@@ -0,0 +1,416 @@
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+// Mostly required this one
+func (c *Commands) AddGroupWithID(ctx context.Context, group *domain.Group, resourceOwner, groupID string) (_ *domain.Group, err error) {
+	ctx, span := tracing.NewSpan(ctx)
+	defer func() { span.EndWithError(err) }()
+	if resourceOwner == "" {
+		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-w8tnSoJxtn", "Errors.ResourceOwnerMissing")
+	}
+	if groupID == "" {
+		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-nDXf5vXoUj", "Errors.IDMissing")
+	}
+
+	existingGroup, err := c.getGroupWriteModelByID(ctx, groupID, resourceOwner)
+	if err != nil {
+		return nil, err
+	}
+	if existingGroup.State != domain.GroupStateUnspecified {
+		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-opamwu", "Errors.Group.AlreadyExists")
+	}
+	group, err = c.addGroupWithID(ctx, group, resourceOwner, groupID)
+	if err != nil {
+		return nil, err
+	}
+	return group, nil
+}
+
+func (c *Commands) AddGroup(ctx context.Context, group *domain.Group, resourceOwner, ownerUserID string) (_ *domain.Group, err error) {
+	if !group.IsValid() {
+		return nil, zerrors.ThrowInvalidArgument(nil, "GROUP-IOGDD", "Errors.Group.Invalid")
+	}
+	if resourceOwner == "" {
+		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-emq7bpQY2s", "Errors.ResourceOwnerMissing")
+	}
+	// Check if the org exists
+	orgWriteModel, err := c.getOrgWriteModelByID(ctx, resourceOwner)
+	if err != nil {
+		return nil, err
+	}
+	if !isOrgStateExists(orgWriteModel.State) {
+		return nil, zerrors.ThrowNotFound(nil, "ORG-1MRds", "Errors.Org.NotFound")
+	}
+	if ownerUserID == "" {
+		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-yf75Gl3Drp", "Errors.OwnerIDMissing")
+	}
+
+	groupID, err := c.idGenerator.Next()
+	if err != nil {
+		return nil, err
+	}
+
+	group, err = c.addGroupWithIDWithOwner(ctx, group, resourceOwner, ownerUserID, groupID)
+	if err != nil {
+		if zerrors.IsErrorAlreadyExists(err) {
+			return nil, zerrors.ThrowAlreadyExists(err, "COMMAND-9f8sJ", "Errors.Group.AlreadyExists")
+		}
+		return nil, err
+	}
+	return group, nil
+}
+
+func (c *Commands) addGroupWithID(ctx context.Context, groupAdd *domain.Group, resourceOwner, groupID string) (_ *domain.Group, err error) {
+	groupAdd.AggregateID = groupID
+	addedGroup := NewGroupWriteModel(groupAdd.AggregateID, resourceOwner)
+	groupAgg := GroupAggregateFromWriteModel(&addedGroup.WriteModel)
+
+	events := []eventstore.Command{
+		group.NewGroupAddedEvent(
+			ctx,
+			groupAgg,
+			groupAdd.Name,
+			groupAdd.Description),
+	}
+	pushedEvents, err := c.eventstore.Push(ctx, events...)
+	if err != nil {
+		return nil, err
+	}
+	err = AppendAndReduce(addedGroup, pushedEvents...)
+	if err != nil {
+		return nil, err
+	}
+	return groupWriteModelToGroup(addedGroup), nil
+}
+
+func (c *Commands) addGroupWithIDWithOwner(ctx context.Context, groupAdd *domain.Group, resourceOwner, ownerUserID, groupID string) (_ *domain.Group, err error) {
```

Comment:
There is an owner mentioned in this function, but I don't see any relevant event created.
Same goes for the `AddGroup` with the attribute `ownerUserID`.
Please remove this functions if there is no owner necessary.